### ISSUE TITLE
Feature request: allow "for collection" to mean "for _ in collection"

### ISSUE THREAD

**Issue by simleb** [JuliaLang/julia#13894]
It would be nice to be able to write:
```julia
[f() for 1:n]
```
instead of
```julia
[f() for _ in 1:n]
```
* * *
Here is some context: I wanted to build a `Vector{Vector{Float64}}` containing empty `Vector{Float64}`s.
`Array(Vector{Vector{Float64}}, n)` doesn't work because the inner vectors are undefined. `repeat(Float64[], outer=[n])` doesn't work because it is just concatenating empty arrays. `collect(repeated(Float64[], n))` works and so does `[Float64[] for _ in 1:n]`, but it could be prettier. That's why I am proposing this teaspoon of syntactic sugar.

Not a game changer but it could be a nice little addition.

<sub>I apologize if this was brought up before. I couldn't find it.</sub>

**Comment by StefanKarpinski** [comment]
I've also often wanted this.

**Comment by johnmyleswhite** [comment]
Although I totally see the appeal of this, I'd like to argue against this change for now since I see this syntax as further exacerbating the existing semantic overlap between comprehensions and `map`.

`[f() for 1:n]` is, AFAICT, exactly equivalent to `map(() -> f(), 1:n)` in theory -- except that, in practice, one has much worse performance. Moreover, both are coupled to type-inference in strange and hard-to-reason about ways, until we introduce type-inference independent comprehensions. (And we also don't have a totally clear story for the return type of `map` yet either.)

To put it another way, I'm pretty worried about us accumulating even more constructs that should be exact synonyms, but aren't really because of the way that some subtle complexities in the language's designs affect both correctness and performance. This kind of stuff makes the language harder to use because these syntactically nice constructs are correlated with complex semantic issues that newcomers can't reason about.

**Comment by stevengj** [comment]
Duplicate of #2675.
